src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp

```diff
@@ -329,50 +349,60 @@ namespace OpenMS
     // Go through all peptides and apply the decoy method to the sequence
     // (pseudo-reverse, reverse or shuffle). Then set the peptides and proteins of the decoy
     // experiment.
+    Size progress = 0;
+    startProgress(0, exp.getPeptides().size(), "Generating decoy peptides");
     for (Size pep_idx = 0; pep_idx < exp.getPeptides().size(); ++pep_idx)
     {
+      setProgress(++progress);
+
       OpenMS::TargetedExperiment::Peptide peptide = exp.getPeptides()[pep_idx];
-      // continue if the peptide has C/N terminal modifications and we should exclude them
-      if (remove_CNterminal_mods && MRMDecoy::has_CNterminal_mods(peptide)) {continue; }
+
       peptide.id = decoy_tag + peptide.id;
-      OpenMS::String original_sequence = peptide.sequence;
+
       if (!peptide.getPeptideGroupLabel().empty())
       {
         peptide.setPeptideGroupLabel(decoy_tag + peptide.getPeptideGroupLabel());
       }
 
       if (method == "pseudo-reverse")
       {
-        peptide = MRMDecoy::pseudoreversePeptide(peptide);
+        // exclude peptide if it has C/N terminal modifications because we can't do a (partial) reverse
+        if (MRMDecoy::has_CNterminal_mods(peptide))
```

Comment:
For the ``pseudo-reverse`` and ``reverse`` methods, any C-/N-terminally modified peptides will be excluded, as they might not be compatible.

Reply:
In the DecoyDatabase, I think we only use reverse. Would it make sense to reuse (or generalize) MRMDecoy to be applicable to non-MRM tools in OpenMS ?

Reply:
There also seems to be a ``shuffle`` method in DecoyDatabase on protein sequence-level. It might be worthwhile to consider adapting the methods also there, however there are important differences. Here, we need to already consider PTMs, whereas with DecoyDatabase, the search engines will take care about this. What might be worthwhile to investigate is to also keep the N- and C-terminus fixed in DecoyDatabase, as this will generate more similar decoys. But this of course depends on the assumptions of the search engines.

Reply:
ok I see.